src/connector/src/enforce_secret_on_cloud.rs

```diff
@@ -27,9 +26,6 @@ pub trait EnforceSecretOnCloud {
     const ENFORCE_SECRET_PROPERTIES_ON_CLOUD: Set<&'static str> = phf_set! {};
 
     fn enforce_secret_on_cloud<'a>(prop_iter: impl Iterator<Item = &'a str>) -> Result<()> {
-        if !Deployment::is_cloud_hosted() {
-            return Ok(());
-        }
```

Comment:
Why to remove this check? (Do you mean to remove the `on_cloud` part from `enforce_secret_on_cloud`?)